src/Templates/src/Microsoft.Maui.Templates.csproj

```diff
@@ -16,10 +15,33 @@
     <NoWarn>$(NoWarn);NU5128</NoWarn>
   </PropertyGroup>
 
-
   <ItemGroup>
     <Content Include="templates\**\*" Exclude="templates\**\bin\**;templates\**\obj\**" />
     <Compile Remove="**\*" />
   </ItemGroup>
 
+  <!-- Eventually replaced by eng/Version.targets -->
+  <Target Name="SetVersions" />
+
+  <Target Name="_UpdateTemplateVersions" BeforeTargets="GenerateNuspec" DependsOnTargets="SetVersions">
+    <PropertyGroup>
+      <PackageReferenceVersion Condition="'$(PackageReferenceVersion)' == ''">$(PackageVersion)</PackageReferenceVersion>
+    </PropertyGroup>
+    <ItemGroup>
+      <_TemplateJsonFile Include="templates\*\.template.config\template.json" />
+      <_TemplateJsonFileWithContent
+          Include="@(_TemplateJsonFile)"
+          Contents="$([System.IO.File]::ReadAllText('%(FullPath)').Replace('MAUI_VERSION_VALUE', '$(PackageReferenceVersion)'))" />
```

Comment:
I'm not sure if this is working. I still see `MAUI_VERSION_VALUE` in the generated template (in the NUPKG, and also once you `dotnet new` it). Not sure what might be wrong here.

The generated template NUPKG has this in template.json:

```json
      "mauiVersion": {
        "type": "parameter",
        "dataType": "string",
        "replaces": "MAUI_VERSION",
        "defaultValue": "MAUI_VERSION_VALUE"
      }
```

Reply:
booo, let me check. Thanks.

Reply:
Seems to have been some task ordering issues. All fixed now.